sway-core/src/semantic_analysis/ast_node/expression/typed_expression_variant.rs

```diff
@@ -559,7 +559,7 @@ impl TypeCheckedStorageAccess {
 }
 
 /// Describes a single subfield access in the sequence when accessing a subfield within storage.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, PartialEq, Eq)]
```

Comment:
I am not sure if this is the right approach. Should I use `look_up_type_id` and implement `PartialEq` myself?

Reply:
I think this perfectly fine!